Modules/Core/Common/include/itkMetaDataDictionary.h

```diff
@@ -81,6 +82,35 @@ class ITKCommon_EXPORT MetaDataDictionary
   // Destructor
   virtual ~MetaDataDictionary();
 
+  /** Returns (metaDataDictionary1 == metaDataDictionary2). */
+  friend bool
+  operator==(const Self & lhs, const Self & rhs)
+  {
+    using KeyValuePair = MetaDataDictionaryMapType::value_type;
+
+    return (lhs.m_Dictionary == rhs.m_Dictionary) ||
+           ((lhs.m_Dictionary != nullptr) && (rhs.m_Dictionary != nullptr)) &&
+             (lhs.m_Dictionary->size() == rhs.m_Dictionary->size()) &&
```

Comment:
`Modules/Core/Common/include/itkMetaDataDictionary.h:93:69: warning: '&&' within '||' [-Wlogical-op-parentheses]`

Since there is only one `||`, maybe you don't need extra parentheses around the second set of conditions?

Reply:
Thanks @dzenanz, hopefully fixed by force-push https://github.com/InsightSoftwareConsortium/ITK/compare/b76ceccb3be9be9b0a016088939ecf7dcf08e50d..ce8cf4336409f4521feae6a6bd76a3420d6545d4